source/portable/os/ota_os_posix.c

```diff
@@ -150,23 +167,42 @@ OtaOsStatus_t Posix_OtaReceiveEvent( OtaEventContext_t * pEventCtx,
     OtaOsStatus_t otaOsStatus = OtaOsSuccess;
     char * pDst = pEventMsg;
     char buff[ MAX_MSG_SIZE ];
+    struct timespec ts = { 0 };
+    time_t currentTime = time( NULL );
 
     ( void ) pEventCtx;
     ( void ) timeout;
 
     /* Receive the next event from OTA event queue.*/
     errno = 0;
 
-    if( mq_receive( otaEventQueue, buff, sizeof( buff ), NULL ) == -1 )
+    /* Set receive timeout. */
+    ts.tv_nsec = OTA_TIME_MS_TO_NS( OTA_TIME_MS_LESS_THAN_S( timeout ) );
+    ts.tv_sec = OTA_TIME_MS_TO_S( timeout );
+
+    /* Detect overflow. */
+    if( ( currentTime > 0 ) && ( ( int64_t ) ( INT32_MAX - ts.tv_sec ) >= ( int64_t ) currentTime ) )
+    {
+        ts.tv_sec += currentTime;
+    }
```

Comment:
if it overflows we should cap to max value?

Reply:
Fixed in new commit.